main.tf

```diff
@@ -19,20 +19,24 @@ locals {
   }
 }
 
+data "aws_ssm_parameter" "ami" {
+  name = "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2"
+}
+
 resource "aws_instance" "bastion" {
-  ami                    = "ami-0c5204531f799e0c6"
+  ami                    = data.aws_ssm_parameter.ami.value
   instance_type          = "t2.micro"
   key_name               = aws_key_pair.key.key_name
   subnet_id              = module.acs["${var.subnet_type}_subnet_ids"][0]
   vpc_security_group_ids = [aws_security_group.sg.id]
-  tags = local.tags
+  tags                   = local.tags
 }
```

Comment:
Do we think we should ignore changes to `ami`? So that the bastion doesn't accidentally get replaced? What's worse, accidentally getting replaced, or an old unpatched instance?

Reply:
Do we anticipate valid reasons for having a bastion long term?

Reply:
Besides convenience, I don't think so.

Reply:
Yeah. If it's part of your system infrastructure, it isn't a bastion.

Reply:
We'll just have to make sure people aren't treating bastions as if they're good places to store SSH keys or other data.

Reply:
I'd prefer erring on the side of security and possibly blowing away bastions rather than keeping unpatched instances around.

Do you think we need to put a large warning in the README?

Reply:
If this is really used as a bastion I don't see why we can't lean towards the best security. But I do agree we should probably have a warning in the README.

Reply:
Done